qcp/QRegister.py

```diff
@@ -0,0 +1,43 @@
+from QMatrices import QMatrices
+import consts
+
+
+class QRegister:
+    def __init__(self, state):
+        """
+        Params
+        ------
+        State: List
+            Binary representation of the state of the QRegister
+        """
+
+        self.initState = state
+        self.stateVec = self.makeStateVec(state)
+
+    def makeStateVec(self, state):
+        stateVec = QMatrices()
+
+        if state[0] == 0:
+            stateVec.setState(consts.ZERO_VECTOR.getState())
```

Comment:
If you add a bit more logic to the `setState()` method to also accept `QMatrix` types, this line could be simplified to
```python
stateVec.setState(const.ZERO_VECTOR)
```
which is more readable (especially if this logic will be repeated).

Reply:
This sounds like a good idea, pardon me for my inadequate knowledge in python coding. Could you kindly demonstrate the code which I should write to implement what you said ?

Reply:
So the `setState()` method is defined at https://github.com/Tiernan8r/quantum_computing_project/pull/17/files#diff-c85dea5e531dbf214a60884e3669b642a1fa5d430f119f5ceb364b75acb872deR11 as
```python
def setState(self, s):
    self.state = s
    return
````
where you're implicitly assuming that `s` is of type list (which is why the `getState()` is called on this line to convert to a list).

If you modify that method definition to be the following:
```python
def setState(self, s):
    if isinstance(s, QMatrices):
        self.state = s.getState()
    elif isinstance(s, list):
        self.state = s
    return
```
Would handle the calling of the `getState()` method within `setState()`, so our code becomes more readable since we can easily see **what** the code is actually doing, without to much clutter.